systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaUtils.java

```diff
@@ -151,4 +154,79 @@ public static void waitForClusterStability(String clusterName) {
+            LOGGER.info("Kafka config before updating '{}'", kafka.getSpec().getKafka().getConfig().toString());
+            Map<String, Object> config = kafka.getSpec().getKafka().getConfig();
+            config.put(kafkaDynamicConfiguration, value);
+            kafka.getSpec().getKafka().setConfig(config);
+            LOGGER.info("Kafka config after updating '{}'", kafka.getSpec().getKafka().getConfig().toString());
+        });
+    }
+
+    /**
+     * Method which, extends the @see updateConfiguration(String clusterName, KafkaConfiguration kafkaConfiguration, Object value) method
+     * with stability and ensures after update of Kafka resource there will be not rolling update
+     * @param clusterName name of the cluster where Kafka resource can be found
+     * @param kafkaDynamicConfiguration key of specific property
+     * @param value value of specific property
+     */
+    public static void  updateConfigurationWithStabilityWait(String clusterName, String kafkaDynamicConfiguration, Object value) {
+        updateSpecificConfiguration(clusterName, kafkaDynamicConfiguration, value);
+        PodUtils.verifyThatRunningPodsAreStable(KafkaResources.kafkaStatefulSetName(clusterName));
+    }
+
+    /**
+     * Method, verifying that updating configuration were successfully changed inside Kafka CR
+     * @param kafkaDynamicConfiguration key of specific property
+     * @param value value of specific property
+     */
+    public static boolean verifyCrDynamicConfiguration(String clusterName, String kafkaDynamicConfiguration, Object value) {
+        LOGGER.info("Dynamic Configuration in Kafka CR is {}={} and excepted is {}={}",
+            kafkaDynamicConfiguration,
+            KafkaResource.kafkaClient().inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getSpec().getKafka().getConfig().get(kafkaDynamicConfiguration),
+            kafkaDynamicConfiguration,
+            value);
+
+        return KafkaResource.kafkaClient().inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getSpec().getKafka().getConfig().get(kafkaDynamicConfiguration).equals(value);
+    }
+
+    /**
+     * Method, which, verifying that updating configuration were successfully changed inside Kafka pods
+     * @param kafkaPodNamePrefix prefix of Kafka pods
+     * @param kafkaDynamicConfiguration key of specific property
+     * @param value value of specific property
+     * @return
+     * true = if specific property match the excepted property
+     * false = if specific property doesn't match the excepted property
+     */
+    public static boolean verifyPodDynamicConfiguration(String kafkaPodNamePrefix, String kafkaDynamicConfiguration, Object value) {
+
+        List<Pod> kafkaPods = kubeClient().listPodsByPrefixInName(kafkaPodNamePrefix);
+
+        for (Pod pod : kafkaPods) {
+
+            TestUtils.waitFor("Wait until dyn.configuration is changed", Constants.GLOBAL_POLL_INTERVAL, CR_CREATION_TIMEOUT,
```

Comment:
Is there any reason for doing this via `kafka-configs.sh` and not just using an `Admin` client instance?

Reply:
AdminClient has one disadvantage. If we use him we need som external listener, which is not a good for run the tests in all envinronments.